src/geojson/geojson.js

```diff
@@ -0,0 +1,494 @@
+        }
+    }
+    let point = "<map-point></map-point>";
+    point = parser.parseFromString(point, "text/html");
+
+    let multiPoint = "<map-multipoint><map-coordinates></map-coordinates></map-multipoint>";
+    multiPoint = parser.parseFromString(multiPoint, "text/html");
+
+    let linestring = "<map-linestring><map-coordinates></map-coordinates></map-linestring>";
+    linestring = parser.parseFromString(linestring, "text/html");
+
+    let multilinestring = "<map-multilinestring></map-multilinestring>";
+    multilinestring = parser.parseFromString(multilinestring, "text/html");
+
+    let polygon = "<map-polygon></map-polygon>";
+    polygon = parser.parseFromString(polygon, "text/html");
+
+    let multiPolygon = "<map-multipolygon></map-multipolygon>";
+    multiPolygon = parser.parseFromString(multiPolygon, "text/html");
+
+    let geometrycollection = "<map-geometrycollection></map-geometrycollection>";
+    geometrycollection = parser.parseFromString(geometrycollection, "text/html");
+
+    let feature = "<map-feature><map-featurecaption></map-featurecaption><map-geometry></map-geometry><map-properties></map-properties></map-feature>";
+    feature = parser.parseFromString(feature, "text/html");
+
+    // Template to add coordinates to Geometries
+    let coords = "<map-coordinates></map-coordinates>";
+    coords = parser.parseFromString(coords, "text/html");
+    
+    //console.log(layer);
+    if (jsonType === "FEATURECOLLECTION") {
+        
+        // Setting bbox if it exists
+        if (json.bbox) {
+            layer.querySelector("layer-").insertAdjacentHTML("afterbegin", "<map-meta name='extent' content='top-left-longitude=" + json.bbox[0] + ", top-left-latitude=" + json.bbox[1] + ", bottom-right-longitude=" + json.bbox[2] + ",bottom-right-latitude=" + json.bbox[3] + "'></map-meta>");
+        }
+
+        let features = json.features;
+        //console.log("Features length - " + features.length);
+        for (let l=0;l<features.length;l++) {
+            geojson2mapml(features[l], options, layer);
+        }
+    } else if (jsonType === "FEATURE") {
+
+        let clone_feature = feature.cloneNode(true);
+        let curr_feature = clone_feature.querySelector('map-feature');
+        
+        // Setting bbox if it exists
+        if (json.bbox) {
+            layer.querySelector("layer-").insertAdjacentHTML("afterbegin", "<map-meta name='extent' content='top-left-longitude=" + json.bbox[0] + ", top-left-latitude=" + json.bbox[1] + ", bottom-right-longitude=" + json.bbox[2] + ",bottom-right-latitude=" + json.bbox[3] + "'></map-meta>");
+        }
+
+        // Setting featurecaption
+        let featureCaption = layer.querySelector("layer-").getAttribute('label');
+        if (typeof options.caption === "function") {
+            featureCaption = options.caption(json);
+        } else if (typeof options.caption === "string") {
+            featureCaption = json.properties[options.caption];

```

Comment:
I think it might be preferable to actually use the string value as the caption in the case where `json.properties[options.caption]` is `undefined`.  Not sure, but 'undefined' as a caption is not useful.

Reply:
Yup good point, better than undefined for sure. Will include this.